python/cudf/cudf/core/dataframe.py

```diff
@@ -842,6 +847,21 @@ def _init_from_list_like(self, data, index=None, columns=None):
         elif len(data) > 0 and isinstance(data[0], pd._libs.interval.Interval):
             data = DataFrame.from_pandas(pd.DataFrame(data))
             self._data = data._data
+        # namedtuple in a list
+        elif (
+            len(data) > 0
+            and columns is None
+            and isinstance(data[0], tuple)
+            and hasattr(data[0], "_fields")
+        ):
+            # pandas behavior is to use the fields from the first tuple
+            # as the column names
+            columns = data[0]._fields
+            values = itertools.zip_longest(*data)
+            data = dict(zip(columns, values))
```

Comment:
And this silently drops columns if the first tuple is shorter than subsequent ones.

Pandas is non-symmetric wrt the handling of named tuples of different lengths. If the first tuple has fewer fields than subsequent ones, we get a `ValueError`, if the first tuple has the same, or more, fields than subsequent ones we get `None`.